### PR TITLE
Feature: note.rs

### PR DESCRIPTION
**Pull request by falk-werner**
This PR introduces a new module: __note.rs__ (finally)
There is no new functionality, just moved the current implementation from main to the new module.

### CODE REVIEW

File: src-tauri/src/note.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+use std::{fs::DirEntry, io::{Error, Result}, path::{Path, PathBuf}};
+use crate::config::{Config};
+
+pub struct Note {
```
Comment (fersaru) [inline src-tauri/src/note.rs:4]: One Note, two notes, all a part of a notebook.
I would like that

**Comment by fersaru** [comment]
One Note, two notes, all a part of a notebook.
I would like that

**Comment by falk-werner** [comment]
> One Note, two notes, all a part of a notebook. I would like that

I renamed the struct to `Notebook`, just to show how this would look like.
Note that this would lead also to renaming of some of the APIs, as `notebook.list()` does not make any sense, since there is only one. It is also very common to use singular in APIs (see [swagger's petstore](https://petstore.swagger.io/)) and also in database tables. _(I took a look at [swaggerhub](https://app.swaggerhub.com/search) and also found quite a number of APIs that uses plural, but it seems to me, that `notes` would be preferred by them.)_

Also note, that I've only applied this change to the backend. When applied to the frontend, it would have impact on `note.js`, since we plan to share the same frontend in the future.

**Comment by fersaru** [comment]
I don't like the idea of changing the frontend so it becomes unique to this note.* implementation.
The "interface" should present the same function signatures as the other backend implementations.
But how the function is implemented should be backend specific.

In the current naming I don't like the `_` in `list_notes` because when I think in objects it reads like there should be a `notes` object.
In objects speaking I thought one notebook has all the configs it needs, it holds a quantity of notes and has the functionalities to present them in a desired format (like a list of only the names). A note would then have knowledge about it's name, it's path and it's tags.

But I think this becomes too complex for the initial idea of having a simple note-tool.
I didn't aim for a discussion about a change in the API with my proposal. I was just thinking "out loud".

I think going back to the "note" naming is closest to the note.* idea, so i would go back to it for now.

**Review by fersaru — APPROVED** [review]

**Pull request merged**